### PR TITLE
Allow _ in Lightning addresses

### PR DESCRIPTION
**Pull request by AaronDewes**
This allows things like example_subdomain.example.com to be used as LN address.

### CODE REVIEW

File: src/common/lib/lnurl.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ const fromInternetIdentifier = (address: string) => {
   // email regex: https://emailregex.com/
```
Comment (escapedcat) [inline src/common/lib/lnurl.ts:16]: I guess the comment should be updated as well then to reflect the improvement?

**Review by reneaaron — APPROVED** [review]
utACK

**Review by escapedcat — APPROVED** [review]
Thanks @AaronDewes !

File: src/common/lib/lnurl.ts [hunk 2]
```diff
@@ -14,9 +14,10 @@ import { bech32Decode } from "../utils/helpers";
 
 const fromInternetIdentifier = (address: string) => {
   // email regex: https://emailregex.com/
+  // modified to allow _ in subdomains
   if (
     address.match(
-      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-_0-9]+\.)+[a-zA-Z]{2,}))$/
```
Comment (bumi) [inline src/common/lib/lnurl.ts:20]: Is a `_` in the spec for domains? I thought that's only letters, digits, and hyphen?

Comment (AaronDewes) [reply src/common/lib/lnurl.ts:20]: It's allowed in subdomains, I think it may not be allowed in second level domains though 🤷‍♂️

Comment (bumi) [reply src/common/lib/lnurl.ts:20]: that regex is too complicated :) 
image

we now match also `_` in the domain. but that's fine :)
thanks for the PR.

**Pull request merged**